static/js/teachers.ts

```diff
@@ -54,18 +54,43 @@ export function rename_class(id: string, class_name_prompt: string) {
     });
 }
 
-export function duplicate_class(id: string, prompt: string, defaultValue: string = '') {
+export function duplicate_class(id: string, second_teacher_prompt: string, prompt: string, defaultValue: string = '') {
+  modal.confirm(second_teacher_prompt, function () {
+    callDuplicateClass(id, prompt, true, defaultValue);
+  }, function () {
+    callDuplicateClass(id, prompt, false, defaultValue);
+  });
+}
+
+function callDuplicateClass(id: string, prompt: string, second_teacher: boolean, defaultValue: string = '') {
```

Comment:
```suggestion
function apiDuplicateClass(id: string, prompt: string, second_teacher: boolean, defaultValue: string = '') {
```
Also, don't forget to change it where it's being called. I do like that you use camel case variables :)) I want to convert all variable/function names in TS to camel case. But that's in the future, and great to start from now!